src/proxy.ts

```diff
@@ -0,0 +1,110 @@
+import { fork, Sequence } from 'effection';
+import { on } from '@effection/events';
+import * as proxy from 'http-proxy';
+import * as http from 'http';
+import { listen, ReadyCallback } from './http';
+import { forkOnEvent,  } from './util';
+import * as trumpet from 'trumpet';
+import * as zlib from 'zlib';
+import { Readable, Writable } from 'stream';
+
+interface ProxyOptions {
+  port: number;
+  targetPort: number;
+  inject?: string;
+};
+
+function pipe(from, to) {
+  fork(function*() {
+    let writer = forkOnEvent(from, 'data', function*(chunk) {
+      to.write(chunk);
+    });
+
+    try {
+      yield on(from, 'end');
+      to.end();
+    } finally {
+      writer.halt();
+    }
+  });
+}
+
+export function* createProxyServer(options: ProxyOptions, ready: ReadyCallback = x => x): Sequence {
+  let proxyServer = proxy.createProxyServer({
+    target: `http://localhost:${options.targetPort}`,
+    selfHandleResponse: true
+  });
+
+  forkOnEvent(proxyServer, 'proxyRes', function*(proxyRes, req, res) {
+    for(let [key, value] of Object.entries(proxyRes.headers)) {
+      res.setHeader(key, value);
+    }
+
+    let contentType = proxyRes.headers['content-type'] as string;
+    let contentEncoding = proxyRes.headers['content-encoding'] as string;
+
+    if(contentType && contentType.split(';')[0] === 'text/html') {
+      res.removeHeader('content-length');
+      res.removeHeader('content-encoding');
+
+      res.writeHead(proxyRes.statusCode, proxyRes.statusMessage);
+
+      let tr = trumpet();
+
+      tr.select('head', (node) => {
+        let rs = node.createReadStream();
+        let ws = node.createWriteStream();
+        ws.write(options.inject || '');
+        rs.pipe(ws);
+      });
+
+      if(contentEncoding && contentEncoding.toLowerCase() == 'gzip') {
+        let zip = zlib.createGunzip();
```

Comment:
A small enough thing, but I'd personally prefer to call this stream `unzip` since that's what happens to bytes that get pushed into it.

Reply:
Fixed :)